### PR TITLE
feat: ensure HashMap computes in the kernel

### PR DESCRIPTION
**Pull request by semorrison**
It seems a poor tradeoff to eke out a little more performance at the cost of losing the ability to compute with `HashMap` in the kernel.

### CODE REVIEW

File: Std/Data/HashMap/Basic.lean [hunk 1]
```diff
@@ -29,18 +29,18 @@ def mk (buckets := 8) (h : 0 < buckets := by decide) : Buckets α β :=
...
     (d : AssocList α β) (h : i.toNat < data.1.size) : Buckets α β :=
-  ⟨data.1.uset i d h, (Array.size_uset ..).symm ▸ data.2⟩
+  ⟨data.1.set ⟨i.toNat, h⟩ d, (Array.size_set ..).symm ▸ data.2⟩
 
 /--
 The number of elements in the bucket array.
 Note: this is marked `noncomputable` because it is only intended for specification.
 -/
-noncomputable def size (data : Buckets α β) : Nat := .sum (data.1.data.map (·.toList.length))
+def size (data : Buckets α β) : Nat := .sum (data.1.data.map (·.toList.length))
```
Comment (JamesGallicchio) [inline Std/Data/HashMap/Basic.lean:40]: per comment in code this was noncomputable to discourage use outside of specification. does it have to be computable?

Comment (semorrison) [reply Std/Data/HashMap/Basic.lean:40]: Oops, sorry, this was from when I was working out what was going on. I'll revert.

**Review by JamesGallicchio — APPROVED** [review]

File: Std/Data/HashMap/Basic.lean [hunk 2]
```diff
@@ -369,3 +369,7 @@ def ofListWith [BEq α] [Hashable α] (l : List (α × β)) (f : β → β → 
     match m.find? p.1 with
     | none   => m.insert p.1 p.2
     | some v => m.insert p.1 <| f v p.2
+
+-- Verify that basic operations are computable in the kernel, and not blocked by
+-- `System.Platform.numBits`.
+example : (HashMap.empty.insert 0 0).isEmpty = false := rfl
```
Comment (digama0) [inline Std/Data/HashMap/Basic.lean:375]: Tests don't go in `Std/`

**Comment by digama0** [comment]
Note, this PR may be rejected or modified pending discussion [on Zulip](https://leanprover.zulipchat.com/#narrow/stream/348111-std4/topic/no.20kernel.20reduction.20of.20HashMap.2Einsert/near/403887177).

**Comment by semorrison** [comment]
Yes, I agree this is a bad idea now. Thanks for the explanations, very helpful.

(Sadly, using RBMap in my application turned out to be even slower that just traversing lists...)

**Pull request closed without merging**